src/orange/transform/Transformation.cc

```diff
@@ -40,14 +40,16 @@ Transformation::Transformation() : Transformation{Translation{}} {}
 /*!
  * Construct and check the input.
  *
- * The input rotation matrix should be an orthonormal matrix without
- * reflecting, thus having a determinant of unity. It is the caller's job to
- * ensure a user-provided low-precision matrix is orthonormalized.
+ * The input rotation matrix should be an orthonormal matrix. Its determinant
+ * is 1 if not reflecting (proper) or -1 if reflecting (improper).  It is the
+ * caller's job to ensure a user-provided low-precision matrix is
+ * orthonormal: see \c celeritas::orthonormalize .
  */
 Transformation::Transformation(Mat3 const& rot, Real3 const& trans)
     : rot_(rot), tra_(trans)
 {
-    CELER_EXPECT(soft_equal(determinant(rot_), real_type(1)));
+    // XXX: should this be a CELER_VALIDATE?
```

Comment:
It is probably a good idea to provide clear warnings or errors even in non-debug mode, in case of interactive changes in GDML files, which is a common use case.

Reply:
My thought was that perhaps this validation should be done "closer" to the user input: for example, Geant4's GDML reader (or transform matrix constructor) executes `rectify` to construct an exact matrix. I think I'll remove the comment.